### ISSUE TITLE
[mitigated] Linux: Ctrl+Shift+E cannot be used as keybinding anymore

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#48480]
Looks like with the update to GTK3 this key combination is reserved for the OS. More info: https://unix.stackexchange.com/questions/394655/ctrlshifte-causes-beeping/394767#394767

**Comment by ykgmfq** [comment]
This will be fixed in the upcoming 3.24 release.
https://gitlab.gnome.org/GNOME/gtk/commit/8be12a2427927cf914c6223c6a2e1267352c40da

**Comment by bpasero** [comment]
@ykgmfq I assume this means users will have to update to this version and there is no other workaround?

**Comment by ykgmfq** [comment]
As far as I know, yes. It won't take long for most distributions to ship the new GTK3 release. Bundling a newer libgtk3 built from source with the snap package might be a solution for Ubuntu LTS users.

**Comment by Tyriar** [comment]
Closing as upstream, nothing more for us to action. Thanks @ykgmfq!

**Comment by costela** [comment]
Workaround for people using traditional (deb|rpm) packages and stuck with older GTK versions:
If your distro uses [ibus](/ibus/ibus) as the default input method (fedora, I believe?), you just have to open `ibus-setup`, go to the "emoji" tab and disable the Ctrl-Shift-e keybinding.
If your distro doesn't use ibus by default (ubuntu, AFAICT), you can either install, enable it globally and use the workaround above, **or** you can start vscode with `env GTK_IM_MODULE=ibus` (or copy the `.desktop` file in the package to `~/.local/share/applications` and change the `Exec` line to include that variable)

**Comment by m-thorsen** [comment]
FYI: the `ibus` package provides `ibus-setup` on Debian/Ubuntu-based distros. This is not installed by default in KDE Neon (although the shortcut is hijacked nonetheless)

**Comment by OnlyLys** [comment]
I used the mitigation strategy by @costela on Manjaro by installing ibus and it works. I could rebind the `Ctrl+Shift+E` to something else.

**Comment by David-Else** [comment]
In Fedora 28 using `ibus-setup` and deleting the short-cut seems to do nothing. Has anyone found any solutions that don't involve system wide changes?

**Comment by OnlyLys** [comment]
@David-Else if I'm not wrong, Fedora 28 uses GNOME correct? `ibus` should work properly with it. At least that's what Arch Wiki says: https://wiki.archlinux.org/index.php/IBus#Input_method_engines. 

When you use `Ctrl+Shift+E` when `ibus` is enabled, do you get some sort of popup box for emojis?

**Comment by David-Else** [comment]
@OnlyLys  @m-thorsen I managed to get it to work with a lot of messing about! I followed the steps once and it did not work, so I did it again and then it worked?! This is a super annoying bug.

What worked was to delete the shortcut in ibus-setup then start vs code with:

`GTK_IM_MODULE=ibus code`

I tried to edit the desktop file but it just stopped appearing, what exact line should I have in there? Cheers. I guess I need to create a special alias for code on the command line to launch code with the added `GTK_IM_MODULE=ibus `code in front unless there is a better way? Thanks guys!

**Comment by OnlyLys** [comment]
@David-Else David, what you're setting with `GTK_IM_MODULE=ibus` is an environment variable. You're setting the `GTK_IM_MODULE` variable to `ibus`. In my opinion, you should have it set globally so when you just call the plain `code` in the terminal, the environment variable will already have the correct value. 

There are a few ways you can do it globally but it depends on what Distro / Desktop Environment you're running. Typically you would hear advice to edit a `~/.profile` or `~/.bash_profile` file or something like that, but your best bet is to Google search the instructions for your distro.

For me for instance, my distro uses KDE and they have provided a really simple way to add environment variables just by [adding a script tile to my configurations folder](https://userbase.kde.org/Session_Environment_Variables).

**Comment by David-Else** [comment]
@OnlyLys Thanks for the info! Yesterday I added:

`alias code="GTK_IM_MODULE=ibus code"`

to my `.bashrc` in Fedora 28 and it seems to work great.

Can you see a downside to doing it this way? ibus is loaded by default, there are no other steps involved.

**Comment by GeorgeSapkin** [comment]
@David-Else what is the complete list of things that you've changed? I've disabled the shortcut in ibus-setup, added the alias and even rebooted and it still doesn't help. Not even when starting code from terminal. I've also tried exporting the variable and changing the desktop file. I'm using default F28 with Gnome.

It is bizarre how this this not considered an issue.

**Comment by David-Else** [comment]
@GeorgeSapkin I just did it on another machine successfully.

1. go to terminal type 'ibus-setup'
2. go to Emoji tab, press the '...' next to Emoji choice to get 'select keyboard shortcut for switching' window
3. use the delete button to delete the shortcut and leave nothing there, press OK
4. Close
5. Try typing `alias code="GTK_IM_MODULE=ibus code"` into the terminal. This should work!
6. If it does just alias this as I put in the post above

It worked first time here in that order, but I do remember it acted funny on the last machine and I had to do it twice... or something like that. I forget.

You are right, this is a massive issues, and it should **at least be mentioned in the readme for the release.**
